pkg/operator/che_cm.go

```diff
@@ -0,0 +1,133 @@
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package operator
+
+import (
+	"github.com/eclipse/che-operator/pkg/util"
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Merge two maps: general Che config and k8s only
+func addMap(a map[string]string, b map[string]string) {
+	for k, v := range b {
+		a[k] = v
+	}
+}
+
+func newCheConfigMap(cheHost string, keycloakURL string) *corev1.ConfigMap {
+	infra := util.GetInfra()
+	cheLabels := map[string]string{"app": "che"}
+	openshiftOAuth := util.GetEnvBool("CHE_OPENSHIFT_OAUTH", false)
+	ingressDomain := util.GetEnv("CHE_INFRA_KUBERNETES_INGRESS_DOMAIN", "")
+	strategy := util.GetEnv("CHE_INFRA_KUBERNETES_SERVER__STRATEGY", "multi-host")
+	workspacesNamespace := namespace
+	tls := "false"
+	openShiftIdentityProviderId := "NULL"
+	if openshiftOAuth {
+		workspacesNamespace = ""
+		openShiftIdentityProviderId = "openshift-v3"
+	}
+	if tlsSupport {
+		protocol = "https"
+		wsprotocol = "wss"
+		tls = "true"
+	}
+	// general envs
+	cheEnv := map[string]string{
+		"CHE_MULTIUSER":          "true",
+		"CHE_HOST":               cheHost,
+		"CHE_PORT":               "8080",
+		"CHE_API":                protocol + "://" + cheHost + "/api",
+		"CHE_WEBSOCKET_ENDPOINT": wsprotocol + "://" + cheHost + "/api/websocket",
+		// todo Make configurable
+		"CHE_DEBUG_SERVER":                              "false",
+		"CHE_INFRASTRUCTURE_ACTIVE":                     infra,
+		"CHE_INFRA_KUBERNETES_BOOTSTRAPPER_BINARY__URL": protocol + "://" + cheHost + "/agent-binaries/linux_amd64/bootstrapper/bootstrapper",
+		"CHE_INFRA_OPENSHIFT_PROJECT":                   workspacesNamespace,
+		"CHE_INFRA_KUBERNETES_PVC_STRATEGY":             pvcStrategy,
+		"CHE_INFRA_KUBERNETES_PVC_QUANTITY":             pvcClaimSize,
+		"CHE_INFRA_OPENSHIFT_TLS__ENABLED":              tls,
+		"CHE_INFRA_KUBERNETES_TRUST__CERTS":             tls,
+		"CHE_JDBC_URL":                                  "jdbc:postgresql://" + postgresHostName + ":" + postgresPort + "/" + chePostgresDb,
+		"CHE_JDBC_USERNAME":                             chePostgresUser,
+		// todo Create a secret for it?
+		"CHE_JDBC_PASSWORD":                             chePostgresPassword,
+		"CHE_LOG_LEVEL":                                 "INFO",
+		"CHE_KEYCLOAK_AUTH__SERVER__URL":                keycloakURL + "/auth",
+		"CHE_INFRA_OPENSHIFT_OAUTH__IDENTITY__PROVIDER": openShiftIdentityProviderId,
+		"CHE_PREDEFINED_STACKS_RELOAD__ON__START":       "true",
+		"CHE_INFRA_KUBERNETES_SERVICE__ACCOUNT__NAME":   "che-workspace",
```

Comment:
Can we use constant value here. Then the same constant may be reused during workspace service account creating https://github.com/eclipse/che-operator/pull/3/commits/673c24c26c5b24cd01c831fba7a9a5f7f9d0ad1c#diff-0b5c445f7c199f4fc2bc74e3cc61cef8R30